qmk_cli/subcommands/console.py

```diff
@@ -141,7 +141,7 @@ def run_forever(self):
                 identifier = (int2hex(message['vendor_id']), int2hex(message['product_id'])) if self.numeric else (message['manufacturer_string'], message['product_string'])
                 message['identifier'] = ':'.join(identifier)
                 message['ts'] = '{style_dim}{fg_green}%s{style_reset_all} ' % (strftime(cli.config.general.datetime_fmt),) if cli.args.timestamp else ''
-
+                message['text'] = message['text'].replace('%', '%%')
                 cli.echo('%(ts)s%(color)s%(identifier)s:%(index)d{style_reset_all}: %(text)s' % message)
```

Comment:
I suppose that the intended way for this to work would be (completely untested):
```suggestion
                cli.echo('%(ts)s%(color)s%(identifier)s:%(index)d{style_reset_all}: %(text)s', **message)
```
(using the formatting capabilities of `cli.echo()` instead of working around them).

However, there is another potential issue here — the formatted text is then passed to [`format_ansi()`](https://github.com/clueboard/milc/blob/6896f13ea66da90c0b5799f81bc7955b9642d608/milc/ansi.py#L44-L56), which replaces substrings like `{fg_red}` with corresponding ANSI escape sequences; this will happen even when such substrings are contained in the messages sent by the keyboard.  But apparently there is no clean way to escape the text to prevent such substitutions and get it printed literally (an unclean way might be to insert a dummy ANSI escape sequence after each `{` character — those escape sequences would then be printed if colors are enabled, or filtered out by the `ansi_escape` regex if colors are disabled).

Reply:
Tried the suggestion, non-functional:
```
Exception in thread Thread-6 (run_forever):
Traceback (most recent call last):
  File "E:/QMK_MSYS/mingw64/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "E:/QMK_MSYS/mingw64/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "E:/QMK_MSYS/mingw64/lib/python3.10/site-packages/qmk_cli/subcommands/console.py", line 145, in run_forever
    cli.echo('%(ts)s%(color)s%(identifier)s:%(index)d{style_reset_all}: %(text)s', **message)
TypeError: MILC.echo() got multiple values for argument 'text'
```

Reply:
I said it was untested :-)

Apparently the presence of the positional argument called `text` in `cli.echo()` interferes with the attempt to pass a keyword argument with the same name; but keyword arguments are the only way to use named format specifiers with that code, because `milc` does not have a piece of code like https://github.com/python/cpython/blob/3.11/Lib/logging/__init__.py#L301-L320 which allows passing a dictionary in a non-keyword argument.  (Ideally `cli.echo()` should use the same conventions as the standard `logging` module, especially because there are also `cli.log.error()` and friends, but…)